src/stores/wordpress-versions-api.ts

```diff
@@ -0,0 +1,153 @@
+		.filter( ( result ) =>
+			isDevelopment ? result.data.response === 'development' : result.data.response === 'autoupdate'
+		)
+		.map( ( { data } ) => {
+			const shortName = extractShortName( data.version );
+			shortNameOccurrences.set( shortName, ( shortNameOccurrences.get( shortName ) || 0 ) + 1 );
+			return {
+				version: data.version,
+				shortName,
+			};
+		} );
+}
+
+function generateVersionLabel(
+	version: string,
+	shortName: string,
+	occurrences: number,
+	isLatest: boolean
+): string {
+	if ( isWordPressDevVersion( version ) ) {
+		return 'nightly';
+	}
+	if ( isLatest ) {
+		return sprintf( __( '%s (latest)' ), version );
+	}
+	if ( occurrences > 1 || isWordPressBetaVersion( version ) ) {
+		return version;
+	}
+	return shortName;
+}
+
+function findLatestStable( versions: ProcessedOffer[] ): ProcessedOffer | undefined {
+	return versions.find(
+		( version: ProcessedOffer ) =>
+			! isWordPressBetaVersion( version.version ) && ! isWordPressDevVersion( version.version )
+	);
+}
+
+export const wordpressVersionsApi = createApi( {
+	reducerPath: 'wordpressVersionsApi',
+	baseQuery: fetchBaseQuery( { baseUrl: 'https://api.wordpress.org' } ),
+	endpoints: ( builder ) => ( {
+		getWordPressVersions: builder.query< WordPressVersion[], void >( {
+			query: () => ( {
+				url: `/core/version-check/1.7/?channel=beta&version=${ MINIMUM_WORDPRESS_VERSION }`,
+			} ),
+			transformResponse: async ( response: unknown ) => {
+				try {
+					const stableData = wordPressApiResponseSchema.parse( response );
+
+					const devResponse = await fetch(
+						'https://api.wordpress.org/core/version-check/1.7/?channel=development'
+					);
+
+					const developmentData = wordPressApiResponseSchema.parse( await devResponse.json() );
+
+					const shortNameOccurrences = new Map< string, number >();
+
+					const stableOffers = processWordPressOffers(
+						stableData.offers,
+						false,
+						shortNameOccurrences
+					);
+
+					const developmentOffer = processWordPressOffers(
+						developmentData.offers,
+						true,
+						shortNameOccurrences
+					)[ 0 ];
+
+					const allOffers = developmentOffer ? [ developmentOffer, ...stableOffers ] : stableOffers;
+					const latestVersion = findLatestStable( allOffers );
+					return allOffers.map( ( { version, shortName } ) => {
+						const isLatest = latestVersion?.version === version;
+						return {
+							isBeta: isWordPressBetaVersion( version ),
+							isDevelopment: isWordPressDevVersion( version ),
+							isLatest,
+							label: generateVersionLabel(
+								version,
+								shortName,
+								shortNameOccurrences.get( shortName ) || 0,
+								isLatest
+							),
+							value: version,
+						};
+					} );
+				} catch ( error ) {
+					if ( error instanceof z.ZodError ) {
+						Sentry.captureException( error );
+					}
+					throw error;
+				}
+			},
+		} ),
+	} ),
+} );
+
+export const { useGetWordPressVersionsQuery } = wordpressVersionsApi;
```

Comment:
I would suggest using a similar wrapper to what we added in the wpcom-api, with the isOffline check, that way the query automatically runs if the connection status changes.
What do you think?

Reply:
Done https://github.com/Automattic/studio/pull/1102/commits/79622b6163d5e286b84b9ecaa8b0bce2a814b9ad and https://github.com/Automattic/studio/pull/1102/commits/7b4cc2e7ee20ec2fdc5396afa3c30604974e2716
I've created a reusable hook, so it's possible to reuse it in any RTK query.